storage/src/main/java/tech/pegasys/teku/storage/client/ChainHead.java

```diff
@@ -62,23 +61,25 @@ public UInt64 findCommonAncestor(final ChainHead other) {
             .minus(spec.getSlotsPerHistoricalRoot(slot));
     while (slot.isGreaterThan(minSlotWithHistoricRoot)) {
       if (getBlockRootAtSlot(slot).equals(other.getBlockRootAtSlot(slot))) {
-        return slot;
+        return new SlotAndBlockRoot(slot, getBlockRootAtSlot(slot));
       }
       slot = slot.minus(1);
     }
     // Couldn't find a common ancestor in the available block roots so fallback to finalized
-    return getState()
-        .getFinalized_checkpoint()
-        .getEpochStartSlot(spec)
-        .min(other.getState().getFinalized_checkpoint().getEpochStartSlot(spec));
+    UInt64 finalizedBlock =
+        getState()
+            .getFinalized_checkpoint()
```

Comment:
Extracting the finalized checkpoint as a variable will make this a lot more readable since it's used three times now.